src/rules/prefer-default-export.js

```diff
@@ -2,15 +2,29 @@
 
 import docsUrl from '../docsUrl';
 
+
+const SINGLE_EXPORT_ERROR_MESSAGE = 'Prefer default export on a file with single export.';
+const ANY_EXPORT_ERROR_MESSAGE = 'Prefer default export to be present on every file that has export.';
+
 module.exports = {
   meta: {
     type: 'suggestion',
     docs: {
       category: 'Style guide',
-      description: 'Prefer a default export if module exports a single name.',
+      description: 'Prefer a default export if module exports a single name or multiple names.',
       url: docsUrl('prefer-default-export'),
     },
-    schema: [],
+    schema: [{
+      type: 'object',
+      properties:{
+        target: { 
+          type: 'string',
+          enum: ['single', 'any'],
+        },
+      },
+      additionalProperties: false,
```

Comment:
i wonder if this is a breaking change - previously, could someone have specified any options they wanted?

Reply:
previously, rule did not have any options so if someone specified any option, that would flag an error. I am wondering if the problem is in using `enum`. However, it works properly on my local machine and flags an error if `target !== 'single && target !== 'any'`

Reply:
I looked at error code for failing test cases. They are all same. The issue was raised for similar error before https://github.com/import-js/eslint-plugin-import/issues/2118.

Reply:
seems like adding `default: 'single'` to the schema solved the issue. All the cases in older versions of `eslint` are passing now

Reply:
I think it as the defaulting on line 36 instead, but glad to hear it's working.

What I'm concerned with, though, is someone passing `[2, { literallyAnything: true }]` to the rule, and previously/currently, it still works - but after this change, eslint would throw an error for this case.

Reply:
I tried to pass random value and it got added to the default `{ target: 'single' }` and error was flagged.

Test case:
```javascript
    test({
      code: `
              export const foo = 'foo';
              export const bar = 'bar'; 
              export default 42;`,
      options: [{
        breakTheRule: 'any',
      }],
    }),
```

error:

```
  1) prefer-default-export valid 
              export const foo = 'foo';
              export const bar = 'bar'; 
              export default 42;:
     Error: rule-tester:
        Configuration for rule "prefer-default-export" is invalid:
        Value {"breakTheRule":"any","target":"single"} should NOT have additional properties.
```

The same error is flagged when I tried to pass random option to other rule: `import/named`:

test case snippet:

```javascript
    test({
      code: 'const { baz } = require("./bar")',
      errors: [error('baz', './bar')],
      options: [{ breakNamedRule: true }],
    }),
```

error message:

```
  1) named valid const { baz } = require("./bar"):
     Error: rule-tester:
        Configuration for rule "named" is invalid:
        Value {"breakNamedRule":true} should NOT have additional properties.
```